wolfssl/wolfcrypt/port/Espressif/esp_crt_bundle.h

```diff
@@ -0,0 +1,222 @@
+ *                       calling process.
+ */
+esp_err_t esp_crt_bundle_attach(void *conf);
+
+
+#if defined(CONFIG_WOLFSSL_CERTIFICATE_BUNDLE) && \
+    defined(CONFIG_WOLFSSL_CERTIFICATE_BUNDLE_DEFAULT_NONE) && \
+    (CONFIG_WOLFSSL_CERTIFICATE_BUNDLE_DEFAULT_NONE == 1)
+
+/* Certificate bundles are enabled, but the "none" option selected */
+
+#else
+/**
+ * @brief      Return ESP_OK for valid bunder, otherwise ESP_FAIL.
+ *
+ * Specific to wolfSSL. Not used by ESP-IDF esp-tls layer.
+ */
+esp_err_t esp_crt_bundle_is_valid();
+
+/**
+ * @brief      Return 1 if Cert Bundle loaded, otheriwse 0.
+ *
+ * Specific to wolfSSL. Not used by ESP-IDF esp-tls layer.
+ */
+int wolfssl_cert_bundle_loaded(void);
+
+/**
+ * @brief      Return 1 is a cert from the bundle was needed
+ *             at connection time, otherwise 0.
+ *
+ * Specific to wolfSSL. Not used by ESP-IDF esp-tls layer.
+ */
+int wolfssl_need_bundle_cert(void);
+
+/**
+ * @brief      Disable and dealloc the certification bundle
+ *
+ * Used by ESP-IDF esp-tls layer.
+ *
+ * Removes the certificate verification callback and deallocates used resources
+ *
+ * @param[in]  conf      The config struct for the SSL connection.
+ */
+void esp_crt_bundle_detach(wolfssl_ssl_config *conf);
+
+/**
+ * @brief      Set the default certificate bundle used for verification
+ *
+ * Used by ESP-IDF esp-tls layer.
+ *
+ * Overrides the default certificate bundle only in case of successful
+ * initialization. In most use cases the bundle should be set through
+ * menuconfig. The bundle needs to be sorted by subject name since binary
+ * search is used to find certificates.
+ *
+ * @param[in]  x509_bundle     A pointer to the certificate bundle.
+ *
+ * @param[in]  bundle_size     Size of the certificate bundle in bytes.
+ *
+ * @return
+ *             - ESP_OK  if adding certificates was successful.
+ *             - Other   if an error occured or an action must be taken
+ *                       by the calling process.
+ */
+esp_err_t esp_crt_bundle_set(const uint8_t *x509_bundle, size_t bundle_size);
+
+
+/**
+ * @brief      Set the issuer and subject values given the current cert.
+ *
+ * Used internally by ESP-IDF esp-tls layer. Also helpful for debugging
+ * and general visibiity to certificate attributes.
+ *
+ * The CERT_TAG can be used at the esp-tls or application layer to indicate
+ * the usage of the respective cert (e.g. the string "peer").
+ *
+ * Turn on WOLFSSL_DEBUG_CERT_BUNDLE to also see ASN1 before/after values.
+ *
+ * @return
+ *             - WOLFSSL_SUCCESS (1)
+ *             - WOLFSSL_FAILURE (0) if unable to get issues and/or subject.
+ */
+int wolfSSL_X509_get_cert_items(char* CERT_TAG,
+                                WOLFSSL_X509* cert,
+                                WOLFSSL_X509_NAME** issuer,
+                                WOLFSSL_X509_NAME** subject);
+
+esp_err_t wolfSSL_bundle_cleanup();
```

Comment:
`esp_err_t wolfSSL_bundle_cleanup(void);`

Reply:
eek, I can't believe I did that again, fixed.